subsys/input/input_longpress.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_input_longpress
+
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(input_longpress, CONFIG_INPUT_LOG_LEVEL);
+
+struct longpress_config {
+	const struct device *input_dev;
+	const uint16_t *input_codes;
+	const uint16_t *short_codes;
+	const uint16_t *long_codes;
+	uint32_t long_delays_ms;
+	uint8_t num_codes;
+};
+
+struct longpress_data_entry {
```

Comment:
Yup good point, added a comment on the entires thing below.

Reply:
BTW I tried using `ARRAY_INDEX()` to avoid the loop but it failed in test, seems like it's only mean to be used for statically allocated arrays.

Reply:
You could store the input-code index in the `longpress_data_entry`, which will consume an extra 4 bytes RAM per input-code.
```
code = cfg->long_codes[entry->index];
```

I think I prefer the explicit index, but I'm ok with the for loop.

Reply:
> You could store the input-code index in the `longpress_data_entry`, which will consume an extra 4 bytes RAM per input-code.

Well there's a bool in that struct and I already bounded this to a 256 limit so an index would come for free. Yeah let's do it.

Reply:
Done, looks nice, no extra ram, thanks for the suggestion, that was a good idea. Kept the index validation just in case, don't see how it could trigger but it shouldn't hurt to have it there.